### ISSUE TITLE
check / adjust behavior of v3 feature production for ROIs with no blob

### ISSUE THREAD

**Issue by hsosik** [hsosik/ifcb-analysis#29]
In my effort to compare classification with v2 and v3 features for a large training set from MVCO, I found what I think is an inconsistent behavior in cases for ROIs with no blob. I think v2 features have an entry for those ROIs with 0s for any feature that requires a blob, but computed values for features that are based on only the raw image. It looks like the test set of v3 features Joe produced have those ROIs skipped altogether. I found this because the change in blob algorithm apparently occasionally produces no blob when previously there was one. Here is an example where there is an old blob (and corresponding v2 features, but no new v3 features:
http://ifcb-data.whoi.edu/mvco/D20170423T045058_IFCB010_02289.html
\\sosiknas1\IFCB_dev\features_v3_test\features\D20170423T045058_IFCB010_fea_v3.csv

**Comment by joefutrelle** [comment]
I think you're just referring to HOG features, right? And the desired fix is to output 0s except for those?

**Comment by joefutrelle** [comment]
Reviewing the code, it looks like it's already designed to produce output with zeros except for image-only features, so there must be a bug in it somewhere.

https://github.com/hsosik/ifcb-analysis/blob/features_v3/python/ifcb_features/all.py#L461

**Comment by joefutrelle** [comment]
I tried two things:

1. computing features for a synthetic noisy flat gray image. produced no blob, and correct feature output.
1. computing features for the image linked to. produced a blob and correct feature output.

this was some sort of bookkeeping error writing out the CSV when I was batch processing. how many ROIs did this happen for?

**Comment by hsosik** [comment]
Hmmm, not many, but I don't know for sure. I changed my script to work around the missing v3 features and I didn't track how many times this occurred. I can probably make a quick script to check when my compilation job is complete later this morning.

**Comment by hsosik** [comment]
I only found two cases: 
    'http://ifcb-data.whoi.edu/mvco/D20170423T045058_IFCB010_02289'
    'http://ifcb-data.whoi.edu/mvco/D20170517T201835_IFCB010_03033'
